newIDE/app/src/AnnouncementsFeed/index.js

```diff
@@ -75,23 +78,46 @@ export const AnnouncementsFeed = ({
     if (!title) {
       const markdownClickableImageRegex = /\[!\[(?<alt>[^\][]*)\]\((?<imageSource>[^\s]*)\)\]\((?<linkHref>[^\s]*)\)/;
       let matches = message.match(markdownClickableImageRegex);
-      if (
-        matches &&
-        matches.groups &&
-        matches.groups.alt &&
-        matches.groups.imageSource &&
-        matches.groups.linkHref
-      ) {
-        return { title, message, isMarkdownImageOnly: true };
+      let groups = matches && matches.groups;
+      if (groups && groups.alt && groups.imageSource && groups.linkHref) {
+        // If the href is pointing to the editor, this can be an internal link,
+        // so we don't open a new link, and instead use the internal router
+        // to navigate to the right page/dialog.
+        if (groups.linkHref.startsWith('https://editor.gdevelop.io')) {
+          const urlParams = new URLSearchParams(
+            groups.linkHref.replace(/.*\?/, '')
+          );
+          // $FlowFixMe - Assume that the arguments are always valid.
+          const route: ?Route = urlParams.get('initial-dialog');
+          const otherParams = {};
+          urlParams.forEach((value, key) => {
+            if (key !== 'initial-dialog') otherParams[key] = value;
+          });
+          if (route) {
+            const onInternalClick = () => navigateToRoute(route, otherParams);
+            const messageWithoutHref = message.replace(
+              markdownClickableImageRegex,
+              `![${groups.alt}](${groups.imageSource})`
+            );
+            return {
+              title,
+              message: messageWithoutHref,
+              isMarkdownImageOnly: true,
+              onInternalClick,
```

Comment:
Could almost be called onClick?

Does not matter what this thing does, it just is called when there is a click

Reply:
yeah